starter-code/src/viking.js

```diff
@@ -1,11 +1,92 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength) {
+        this.health = health;
+        this.strength = strength;
+    }
+
+    attack() {
+        return this.strength;
+    }
+
+    receiveDamage(damage) {
+        this.health = this.health - damage;
+    }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor(name, health, strength) {
+        super(health, strength);
+        this.name = name
+    }
+
+    receiveDamage(damage) {
+        super.receiveDamage(damage)
+        return this.health > 0 ? `${this.name} has received ${damage} points of damage` : `${this.name} has died in act of combat`;
+    }
+
+    battleCry() {
+        return "Odin Owns You All!";
+    }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+    constructor(health, strength) {
+        super(health, strength);
+    }
+
+    receiveDamage(damage) {
+        super.receiveDamage(damage)
+        return this.health > 0 ? `A Saxon has received ${damage} points of damage` : `A Saxon has died in combat`;
+    }
+}
 
 // War
-class War {}
+class War {
+    constructor() {
+        this.vikingArmy = [];
+        this.saxonArmy = [];
+    }
+
+    addViking(viking) {
+        this.vikingArmy.push(viking);
+    }
+
+    addSaxon(saxon) {
+        this.saxonArmy.push(saxon);
+    }
+
+    vikingAttack() {
+        let randomNumberSaxon = Math.floor(Math.random() * this.saxonArmy.length);
+        var randomSaxon = this.saxonArmy[randomNumberSaxon];
+        var randomViking = this.vikingArmy[Math.floor(Math.random() * this.vikingArmy.length)];
+        if (randomViking.strength >= randomSaxon.health) {
+            this.saxonArmy.splice(randomNumberSaxon, 1);
```

Comment:
Normally here you are supposed to call receiveDamage(damage) before removing the dead solider form the array...
To do that you could hold your function inside a variable (that will at the same time call your function and save the result) and then return this variable at the end.